lib/authentification/authentification_base/lib/src/models/demo_user.dart

```diff
@@ -6,9 +6,19 @@
 //
 // SPDX-License-Identifier: EUPL-1.2
 
+import 'dart:developer';
+
 import 'package:authentification_base/authentification.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:firebase_auth_platform_interface/src/auth_provider.dart';
 
 class DemoUser extends AuthUser {
+  DemoUser()
+      : super(
+          uid: 'testUser',
+          firebaseUser: DemoFirebaseUser(),
```

Comment:
Not sure how we should handle this the `DemoUser`. Because we are forced to provider a `firebaseUser` or restructure the user (like making the `firebaseUser` null). The `User` of the Firebase Auth package has no constructor so we need to create our own `DemoFirebaseUser` which brings the problem of implementing methods like `delete()`